helper-script

```diff
@@ -0,0 +1,270 @@
+            ;;
+        m)	moddir=${OPTARG}
+            ;;
+        n)  accountname=${OPTARG}
+            ;;
+        p)  protondir=${OPTARG}
+            ;;
+        x)  prefixdir=${OPTARG}
+            ;;
+        h|\?)
+            usage
+            exit 0
+            ;;
+        a)  ats=true
+            ;;
+        e)  ets2=true
+            ;;
+        s)  startgame=true
+            ;;
+        u)  update=true
+            ;;
+        esac
+    done
+
+    # restore all unused arguments
+    shift $((OPTIND-1))
+    [ "${1:-}" = "--" ] && shift
+}
+
+# fallback to old local folder
+if [ -z "$moddir" ] && [ -d "$scriptdir/truckersmp" ]; then
+    moddir="$scriptdir/truckersmp"
+elif [ -z "$moddir" ]; then
+    moddir="$XDG_CACHE_HOME/truckersmp-cli"
+fi
+if [ -z "$protondir" ]; then
+    protondir=$(find_proton)
+fi
+
+# check for errors
+{
+# what to do?
+if ! [ "$update" = true ] && ! [ "$startgame" = true ]; then
+    >&2 echo "Update (-u) and/or start the game (-s)?"
+fi
+
+# need account to update
+if [ "$update" = true ] && [ -z "$accountname" ]; then
+    >&2 echo "Need the steam account name to update. (-n name)"
+    exit 1
+fi
+
+# only one game at a time
+if [ "$ats" = true ] && [ "$ets2" = true ]; then
+    >&2 echo "It's only possible to use one game at a time."
+    exit 1
+elif ! [ "$ats" = true ] && ! [ "$ets2" = true ]; then
+    >&2 echo "Need at least one game. ATS (-a) or ETS2 (-e)"
+    exit 1
+elif [ "$ats" = true ]; then
+    steamid="270880"
+    if [ -z "$prefixdir" ]; then
+        prefixdir="$XDG_DATA_HOME/truckersmp-cli/American Truck Simulator/prefix"
+    fi
+    if [ -z "$gamedir" ]; then
+        gamedir="$XDG_DATA_HOME/truckersmp-cli/American Truck Simulator/data"
+    fi
+elif [ "$ets2" = true ]; then
+    steamid="227300"
+    if [ -z "$prefixdir" ]; then
+        prefixdir="$XDG_DATA_HOME/truckersmp-cli/Euro Truck Simulator 2/prefix"
+    fi
+    if [ -z "$gamedir" ]; then
+        gamedir="$XDG_DATA_HOME/truckersmp-cli/Euro Truck Simulator 2/data"
+    fi
+fi
+
+echo "Using $steamid"
+echo "GAMEDIR: $gamedir"
+echo "prefixdir: $prefixdir"
+echo "protondir: $protondir"
+
+# check $protondir
+if [ "$startgame" = true ] &&  ! [ -f "$protondir/proton" ]; then
+    >&2 echo "Proton not found."
+    >&2 echo    "Make sure to start at least one game with proton manually or" \
+                "set the correct path with -p"
+    exit 1
+fi
+
+# check for game if no update
+if  ! [ "$update" = true ] \
+    && ! [ -f "$gamedir/bin/win_x64/eurotrucks2.exe" ] \
+    && ! [ -f "$gamedir/bin/win_x64/amtrucks.exe" ]; then
+    >&2 echo "No update wanted and game not found."
+    exit 1
+fi
+}
+
+# download ATS/ETS2
+if [ "$update" = true ]; then
+    # make sure steam is closed before updating
+    # it's possible to update with the steam client open but the client looses
+    # all connectivity and asks for password and steam guard code after restart
+    while pidof steam; do
+	    pkill steam
```

Comment:
pkill does fuzzy matching on it's argument (try `pgrep fire` and you'll catch firefox's PID)
```suggestion
	    pkill -x steam
```
would be a good start, though I'm not sure the steam process is the only one you have to kill in order to terminate Steam (I have steam.sh in mind for example).

Reply:
You're right, that's better. Closing `steam` will also close all it's children so this should be fine. Don't know what `steam.sh` you're talking about. My main steam process is just `steam` alone.

Reply:
For the record: Steam also writes it's current pid into `$HOME/.steam/registry.vdf`.